snuba/web/rpc/v1/resolvers/R_eap_items/resolver_trace_item_table.py

```diff
@@ -0,0 +1,363 @@
+            else attribute_key_to_expression,
+        )
+        # aggregation label may not be set and the column label takes priority anyways.
+        function_expr = replace(function_expr, alias=column.label)
+        return function_expr
+    elif column.HasField("formula"):
+        formula_expr = _formula_to_expression(column.formula, request_meta)
+        formula_expr = replace(formula_expr, alias=column.label)
+        return formula_expr
+    elif column.HasField("literal"):
+        return literal(column.literal.val_double)
+    else:
+        raise BadSnubaRPCRequestException(
+            "Column is not one of: aggregate, attribute key, or formula"
+        )
+
+
+def build_query(request: TraceItemTableRequest) -> Query:
+    if use_eap_items_table(request.meta):
+        entity = Entity(
+            key=EntityKey("eap_items"),
+            schema=get_entity(EntityKey("eap_items")).get_data_model(),
+            sample=None,
+        )
+    else:
+        entity = Entity(
+            key=EntityKey("eap_spans"),
+            schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+            sample=None,
+        )
+
+    selected_columns = []
+    for column in request.columns:
+        # The key_col expression alias may differ from the column label. That is okay
+        # the attribute key name is used in the groupby, the column label is just the name of
+        # the returned attribute value
+        selected_columns.append(
+            SelectedExpression(
+                name=column.label,
+                expression=_column_to_expression(column, request.meta),
+            )
+        )
+        selected_columns.extend(_get_reliability_context_columns(column, request.meta))
+
+    item_type_conds = (
+        [f.equals(snuba_column("item_type"), request.meta.trace_item_type)]
+        if use_eap_items_table(request.meta)
+        else []
+    )
+    res = Query(
+        from_clause=entity,
+        selected_columns=selected_columns,
+        condition=base_conditions_and(
+            request.meta,
+            trace_item_filters_to_expression(
+                request.filter,
+                attribute_key_to_expression_eap_items
+                if use_eap_items_table(request.meta)
+                else attribute_key_to_expression,
+            ),
+            *item_type_conds,
+        ),
+        order_by=_convert_order_by(request.order_by, request.meta),
+        groupby=[
+            attribute_key_to_expression_eap_items(attr_key)
+            if use_eap_items_table(request.meta)
+            else attribute_key_to_expression(attr_key)
+            for attr_key in request.group_by
+        ],
+        # Only support offset page tokens for now
+        offset=request.page_token.offset,
+        # protobuf sets limit to 0 by default if it is not set,
+        # give it a default value that will actually return data
+        limit=request.limit if request.limit > 0 else _DEFAULT_ROW_LIMIT,
```

Comment:
nit: consider logging `limit` so we would know when falling back to the default.